numpy/core/src/umath/ufunc_type_resolution.c

```diff
@@ -325,11 +325,19 @@ PyUFunc_SimpleBinaryComparisonTypeResolver(PyUFuncObject *ufunc,
                 ufunc_name);
         return -1;
     }
-
     /*
      * Use the default type resolution if there's a custom data type
      * or object arrays.
      */
+    if (PyArray_ISCOMPLEX(operands[0]) &&
+        PyArray_ISCOMPLEX(operands[1]) &&(
```

Comment:
This should apply if even one of the operands is complex.

Reply:
I agree, will make the change.